src/__tests__/index.js

```diff
@@ -100,3 +101,17 @@ test("works with devtools store enhancer", () => {
     store.dispatch({ type: "SOME_ACTION" });
   }).not.toThrow();
 });
+
+// there were some reports that this might not be working correctly
+test("coming online processes outbox", () => {
+  const { middleware, enhanceReducer } = createOffline(defaultConfig);
+  const reducer = enhanceReducer(defaultReducer);
+  const store = createStore(reducer, applyMiddleware(middleware));
+
+  const action = { type: "REQUEST", meta: { offline: { effect: {} } } };
+  store.dispatch(action);
+  expect(defaultConfig.effect).not.toBeCalled();
+
+  store.dispatch(networkStatusChanged(true));
```

Comment:
I’m not sure this is correct, the defaultReducer has the online prop as true. Therefore we first need to set it as false, then dispatch the effect, and finally dispatch again the networkStatusChanged as true.

Reply:
@sorodrigo It does set `offline` properly. The default from `offlineUpdater`  overwrites the default from `defaultReducer`, so it is initialized to `false`. It's super confusing though and should be changed.

I'll make the changes tomorrow.